src/source/test_waveform.rs

```diff
@@ -0,0 +1,196 @@
+//! Generator sources for various periodic test waveforms.
+//!
+//! This module provides several periodic, deterministic waveforms for testing other sources and
+//! for simple additive sound synthesis. Every source is monoaural and in the codomain `[-1.0f32,
+//! 1.0f32]` 
+//!
+//! # Example
+//!
+//! ```
+//! use rodio::source::{TestWaveform,TestWaveformFunction};
+//!
+//! let tone = TestWaveform::new(cpal::SampleRate(48000), 440.0, TestWaveformFunction::Sine);
+//! ```
+use std::f32::consts::TAU;
+use std::time::Duration;
+
+use super::SeekError;
+use crate::Source;
+
+/// Test waveform functions.
+#[derive(Clone, Debug)]
+pub enum TestWaveformFunction {
+    /// A sinusoidal waveform.
+    Sine,
+    /// A triangle wave.
+    Triangle,
+    /// A square wave, rising edge at t=0.
```

Comment:
does it matter where the rising edge is? The more specific we document the behavior the less flexible the implementation can be. If we say the rising edge is at t=0 then one user can start depending on that and we can never change it without their code breaking. 

I do not know enough about audio to know if its useful to specify where the rising edge starts. If this is something that is needed then I am all for documenting it.